waku/v2/peermanager/topic_event_handler_test.go

```diff
@@ -0,0 +1,297 @@
+	relays := make([]*relay.WakuRelay, 5)
+
+	for i := 0; i < 5; i++ {
+		relays[i], hosts[i], _ = makeWakuRelay(t, log)
+		err := relays[i].Start(ctx)
+		require.NoError(t, err)
+	}
+
+	// Create peer manager instance with the first hosts
+	pm, eventBus := makePeerManagerWithEventBus(t, relays[0], &hosts[0])
+	pm.ctx = ctx
+	pm.RegisterWakuProtocol(relay.WakuRelayID_v200, relay.WakuRelayENRField)
+
+	// Connect host[0] with all other hosts to reach 4 connections
+	for i := 1; i < 5; i++ {
+		pm.host.Peerstore().AddAddrs(hosts[i].ID(), hosts[i].Addrs(), peerstore.PermanentAddrTTL)
+		err := pm.host.Connect(ctx, hosts[i].Peerstore().PeerInfo(hosts[i].ID()))
+		require.NoError(t, err)
+		err = pm.host.Peerstore().(wps.WakuPeerstore).SetDirection(hosts[i].ID(), network.DirOutbound)
+		require.NoError(t, err)
+
+	}
+
+	go pm.connectivityLoop(ctx)
+
+	time.Sleep(2 * time.Second)
+
+	if len(pm.host.Peerstore().(*wps.WakuPeerstoreImpl).PeersByPubSubTopic(pubSubTopic)) == 0 {
+		log.Info("No peers for the topic yet")
+	}
+
+	// Subscribe to Pubsub topic
+	for i := 0; i < 5; i++ {
+		_, err := relays[i].Subscribe(ctx, protocol.NewContentFilter(pubSubTopic))
+		require.NoError(t, err)
+	}
+
+	// peerEvt to find: relay.PEER_JOINED, relay.PEER_LEFT
+	peerEvt := relay.EvtPeerTopic{
+		PubsubTopic: pubSubTopic,
+		PeerID:      hosts[1].ID(),
+		State:       relay.PEER_JOINED,
+	}
+
+	emitter, err := eventBus.Emitter(new(relay.EvtPeerTopic))
+	require.NoError(t, err)
+
+	err = emitter.Emit(peerEvt)
+	require.NoError(t, err)
+
+	// Process subscribe event and first PEER_JOINED event
+	for i := 0; i < 2; i++ {
+		select {
+		case e := <-pm.sub.Out():
+			switch e := e.(type) {
+			case relay.EvtPeerTopic:
+				{
+					log.Info("Handling topic event...")
+					peerEvt := (relay.EvtPeerTopic)(e)
+					pm.handlerPeerTopicEvent(peerEvt)
+					for _, peer := range pm.host.Peerstore().(*wps.WakuPeerstoreImpl).PeersByPubSubTopic(pubSubTopic) {
+						log.Info("hosts before", zap.String("peer", peer.String()))
+						log.Info("peer", zap.String("connectedness", string(rune(pm.host.Network().Connectedness(peer)))))
+					}
+
+				}
+			case relay.EvtRelaySubscribed:
+				{
+					log.Info("Handling subscribe event...")
+					eventDetails := (relay.EvtRelaySubscribed)(e)
+					pm.handleNewRelayTopicSubscription(eventDetails.Topic, eventDetails.TopicInst)
+				}
+			default:
+				require.Fail(t, "unexpected event arrived")
+			}
+
+		case <-ctx.Done():
+			require.Fail(t, "closed channel")
+		}
+	}
+
+	// Evaluate topic health - unhealthy at first, because no peers connected
+	peerTopic, ok := pm.subRelayTopics[pubSubTopic]
+	if ok {
+		log.Info("New topic subscribed", zap.String("topic", peerTopic.topic.String()))
+	}
+	pm.checkAndUpdateTopicHealth(peerTopic)
+	require.Equal(t, TopicHealth(UnHealthy), peerTopic.healthStatus)
+	time.Sleep(2 * time.Second)
+
+	// Process second to fourth PEER_JOINED event
+	for i := 2; i < 4; i++ {
+
+		peerEvt2 := relay.EvtPeerTopic{
```

Comment:
Similar comment to above, no need to emit events rather just let peers subscribe to a topic.

Reply:
I've simplified event handling test to focus on adding peers to topic and connectedness only. Getting to health  would require full stack and this could be tested better at interop level.